src/components/GraphFilter/GraphFilter.tsx

```diff
@@ -131,9 +131,11 @@ export default class GraphFilter extends React.Component<GraphFilterProps, Graph
               <Button onClick={this.props.onLegend}>{this.props.hideLegend ? 'Show' : 'Hide'} Legend</Button>
             </div>
             <div className="form-group">
-              <Button disabled={this.props.disabled} onClick={this.handleRefresh}>
-                <Icon name="refresh" />
-              </Button>
+              <Spinner loading={this.props.disabled}>
```

Comment:
Is the spinner over the refresh button really back?

Reply:
This is a toolbar not a status area.

Reply:
Having multiple areas 'spin' is not really a good user experience to me.

Reply:
This is just for the transition, I need to have a spinner in the meantime. The PR for the loading in the header was reverted and will put back after this PR

Reply:
There is only one Spinner this time, but will go away in the next PR.